File: csharp/CsSearch/CsSearch/EmbeddedResource.cs
using System;
using System.IO;
using System.Reflection;
using System.Text;

namespace CsSearch
{
    public static class EmbeddedResource
    {
        public static string GetResourceFileContents(string namespaceAndFileName)
        {
            try
            {
                using (var stream = typeof(EmbeddedResource).GetTypeInfo().Assembly.GetManifestResourceStream(namespaceAndFileName))
                using (var reader = new StreamReader(stream, Encoding.UTF8))
                    return reader.ReadToEnd();
            }

            catch(Exception)
            {
                //ApplicationProvider.WriteToLog<EmbeddedResource>().Error(exception.Message);
                throw new Exception($"Failed to read Embedded Resource {namespaceAndFileName}");
            }
        }
    }
}

File: fsharp/FsSearch/FsSearch/EmbeddedResource.fs
namespace FsSearch

open System.IO
open System.Reflection

module EmbeddedResource = 

    let GetResourceFileContents (namespaceAndFileName : string) : string = 
        let contents =
            try
                use stream = Assembly.GetExecutingAssembly().GetManifestResourceStream(namespaceAndFileName)
                use reader = new StreamReader(stream)
                reader.ReadToEnd()
            with
            | :? IOException as e -> printfn "%s" e.Message; ""
        contents
